src/python/grpcio/grpc/_server.py

```diff
@@ -1172,15 +1205,23 @@ def _process_event_and_continue(
             state.due.remove(_SHUTDOWN_TAG)
             if _stop_serving(state):
                 should_continue = False
-    elif event.tag is _REQUEST_CALL_TAG:
+    elif (
+        event.tag is _REQUEST_CALL_TAG
+        or event.tag in state.registered_method_handlers.keys()
+    ):
+        registered_method_name = None
+        if event.tag in state.registered_method_handlers.keys():
+            registered_method_name = event.tag
         with state.lock:
-            state.due.remove(_REQUEST_CALL_TAG)
+            state.due.remove(event.tag)
             concurrency_exceeded = (
                 state.maximum_concurrent_rpcs is not None
                 and state.active_rpc_count >= state.maximum_concurrent_rpcs
             )
             rpc_state, rpc_future = _handle_call(
                 event,
+                registered_method_name,
```

Comment:
I think we should handle both case with the same `_handle_call` and `_find_method_handler` function because `_handle_call` also takes care of exception handling and `_find_method_handler` takes care of interceptors, both of them works the same for registered and unregistered calls.

Agree that we can refactor them to reduce the number of conditions, let me see what I can do.